ziti/edge/impl/conn.go

```diff
@@ -507,32 +514,105 @@ func (conn *edgeConn) newChildConnection(message *channel2.Message) {
-			} else {
-				newConnLogger.Debug("tx crypto established")
-			}
-		}
-
-		listener.acceptC <- edgeCh
-	} else {
-		logger.Errorf("failed to receive start after dial. got %v", startMsg)
-	}
+	listener.acceptC <- edgeCh
 }
 
 func (conn *edgeConn) GetAppData() []byte {
 	return conn.appData
 }
+
+func (conn *edgeConn) CompleteAcceptSuccess() error {
+	if conn.acceptCompleteHandler != nil {
+		result := conn.acceptCompleteHandler.dialSucceeded()
+		conn.acceptCompleteHandler = nil
+		return result
+	}
+	return nil
+}
+
+func (conn *edgeConn) CompleteAcceptFailed(err error) {
+	if conn.acceptCompleteHandler != nil {
+		conn.acceptCompleteHandler.dialFailed(err)
+		conn.acceptCompleteHandler = nil
+	}
+}
+
+type newConnHandler struct {
+	conn                 *edgeConn
+	edgeCh               *edgeConn
+	message              *channel2.Message
+	txHeader             []byte
+	routerProvidedConnId bool
+}
+
+func (self *newConnHandler) dialFailed(err error) {
+	token := string(self.message.Body)
+	logger := pfxlog.Logger().WithField("connId", self.conn.Id()).WithField("token", token)
+
+	newConnLogger := pfxlog.Logger().
+		WithField("connId", self.edgeCh.Id()).
+		WithField("parentConnId", self.conn.Id()).
+		WithField("token", token)
+
+	newConnLogger.Errorf("Failed to establish connection: %v", err)
+	reply := edge.NewDialFailedMsg(self.conn.Id(), err.Error())
+	reply.ReplyTo(self.message)
+	if err := self.conn.SendPrioritizedWithTimeout(reply, channel2.Highest, time.Second*5); err != nil {
+		logger.Errorf("Failed to send reply to dial request: (%v)", err)
+	}
+}
+
+func (self *newConnHandler) dialSucceeded() error {
+	token := string(self.message.Body)
+	logger := pfxlog.Logger().WithField("connId", self.conn.Id()).WithField("token", token)
+
+	newConnLogger := pfxlog.Logger().
+		WithField("connId", self.edgeCh.Id()).
+		WithField("parentConnId", self.conn.Id()).
+		WithField("token", token)
+
+	newConnLogger.Debug("new connection established")
+
+	reply := edge.NewDialSuccessMsg(self.conn.Id(), self.edgeCh.Id())
+	reply.ReplyTo(self.message)
+
+	if !self.routerProvidedConnId {
+		startMsg, err := self.conn.SendPrioritizedAndWaitWithTimeout(reply, channel2.Highest, time.Second*5)
+		if err != nil {
+			logger.Errorf("Failed to send reply to dial request: (%v)", err)
+			return err
+		}
+
+		if startMsg.ContentType != edge.ContentTypeStateConnected {
+			logger.Errorf("failed to receive start after dial. got %v", startMsg)
+			return errors.Errorf("failed to receive start after dial. got %v", startMsg)
+		}
+	} else if err := self.conn.SendPrioritizedWithTimeout(reply, channel2.Highest, time.Second*5); err != nil {
+		logger.Errorf("Failed to send reply to dial request: (%v)", err)
+		return err
+	}
+
+	if self.txHeader != nil {
+		newConnLogger.Debug("sending crypto header")
+		if _, err := self.edgeCh.MsgChannel.Write(self.txHeader); err != nil {
+			newConnLogger.Errorf("failed to write crypto header: %v", err)
```

Comment:
I don't know if you are trying to standardized error output but this one is `%v` and not `(%v)`

Reply:
Did some tidying in the file. Where possible I used WithError instead of Errorf